### ISSUE TITLE
Random bookmarks showing while not logged in

### ISSUE THREAD

**Issue by NiallJoeMaher** [codu-code/codu#416]
## Context

I haven't explored, but I have noticed random filled bookmarks on the homepage while logged out. 

<img width="1202" alt="Screenshot 2023-09-14 at 19 04 16" src="https://github.com/codu-code/codu/assets/12615742/25e53716-47aa-408f-a5ec-e984a140d8f8">

## Expected Behavior

There should be no bookmarks for people that are logged out since they don't have an account to bookmark things. 

## Current Behavior

Crazy stuff, go to the homepage logged out and you'll see there are bookmarks selected (they are blue when selected).

**Comment by John-Paul-Larkin** [comment]
My first attempt at open source contribution, so bear with me as I give this issue far too much attention. 
As it only occurs when logged out, I think it can be easily fixed by checking for an active session, before setting the color of the bookmark, in the ArticlePreview.tsx component.  
className={`w-6 h-6${session && bookmarked ? " fill-blue-400" : ""}`}
If you are happy with that, let me know and I will submit a pull request.  
 
However, this does not get to the bottom of the problem, so I will keep looking.
1. Almost all posts are bookmarked on the homepage, apart from the post “New feature alert -  Notifications”
2. If the post is bookmarked on the homepage, it will also be bookmarked on the /articles page, but when you click through to the individual articles page, it will not be bookmarked there.
3. The issue is not recreated on localhost.

**Comment by John-Paul-Larkin** [comment]
Actually this can be repeatable on localhost. What is happening is that any article that is liked by any logged in user, has the users id added to the Bookmarks array, which is a property of the Post object. When prisma queries the DB, it selects the userId from Bookmarks where userId.
 bookmarks: {
          select: { userId: true },
          where: { userId: userId },
 },
This seems to work fine when the user is logged in. However when userId is undefined it returns all the userId. This seems illogical to me, but I have not used prisma before, so I'm off to read the docs. 
I think this is what is happening, but I can't test it exactly, as I am the sole user on localhost.

I added the following line to the cleaned functions in the ‘randomTrending’ and ‘all’ queries.
if (userId === undefined) currentUserLikesPost = false;
I submitted a pull request, but plan to play around with the code a bit more.
